lib/facter/resolvers/processors_lscpu.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Linux
+      class ProcessorsLscpu < BaseResolver
+        init_resolver
+
+        ITEMS = { threads_per_core: "| grep -e 'Thread(s)'",
+                  cores_per_socket: "-e 'Core(s)'" }.freeze
+
+        class << self
+          #:cores_per_socket
+          #:threads_per_core
+
+          private
+
+          def post_resolve(fact_name, _options)
+            @fact_list.fetch(fact_name) { read_cpuinfo(fact_name) }
+          end
+
+          def read_cpuinfo(fact_name)
+            lscpu_output = Facter::Core::Execution.execute("lscpu #{ITEMS.values.join(' ')}", logger: log)
```

Comment:
I think the initial comment suggested to have uniform values in ITEMS, something like
```
        ITEMS = { threads_per_core: "-e 'Thread(s)'",
                         cores_per_socket: "-e 'Core(s)'" }.freeze
```
and you could have here `lscpu | grep #{ITEMS.values.join(' ')}` instead `lscpu #{ITEMS.values.join(' ')}`